crates/matrix-sdk-crypto/src/verification/cache.rs

```diff
@@ -183,15 +194,28 @@ impl VerificationCache {
         recipient: &UserId,
         recipient_device: &DeviceId,
         content: OutgoingContent,
+        request_info: Option<RequestInfo>,
     ) {
+        let request_id = if let Some(request_info) = request_info {
+            trace!(
+                %recipient,
+                ?request_info,
+                "Storing the request info, waiting for the request to be marked as sent"
+            );
+
+            self.flow_ids_waiting_for_response.insert(
+                request_info.request_id.to_owned(),
+                (recipient.to_owned(), request_info.flow_id),
+            );
+            request_info.request_id
+        } else {
+            TransactionId::new()
```

Comment:
I wonder if, semantically speaking, it should not be better to call `TransactionId::default` here. In both cases, it will create an empty transaction ID. Just wondering.

Reply:
I don't think `TransactionId` implements `Default`, so we can't do that.

Reply:
:+1: